module/zfs/spa.c

```diff
@@ -4570,6 +4570,7 @@ spa_export_common(char *pool, int new_state, nvlist_t **oldconfig,
 	if ((new_state == POOL_STATE_DESTROYED) ||
 	    (new_state == POOL_STATE_EXPORTED)) {
 		uzfs_zvol_destroy_cb(spa_name(spa), NULL);
+		uzfs_close_pool(spa);
```

Comment:
is this related to 'uzfs_open_pool' in zrepl_import? does this mean - if any pool/volume created from CLI will get exported and destroyed?

Reply:
I mean - if I understood correctly, without this uzfs_close_pool, if any pool/volume created from CLI will get exported and destroyed.

Reply:
We need to open pool uzfs_open_pool() and we need to close them using uzfs_close_pool() if we dont close it here, we keep the UZFS_SPA_TAG held.

Reply:
This is needed because our pool open has pre/post hooks and puts a hold on the spa

Reply:
I want to check whether all the cases of pool export / destroy is fixed.
Two cases that I'm aware are:
1. import pool using 'zrepl import -t <ip>'
2. created pool using 'zpool create ..' after starting zrepl as 'zrepl start -t <ip>'
I understand change is needed.. :) I can see where case 1 being fixed.
Does case 2 also have issue and got fixed with this? or case 2 was already working?